src/main/java/org/skriptlang/skript/registration/SyntaxRegister.java

```diff
@@ -14,13 +15,44 @@
  */
 final class SyntaxRegister<I extends SyntaxInfo<?>> {
 
+	private static int calculateComplexityScore(SyntaxInfo<?> info) {
+		return info.patterns().stream()
+				.mapToInt(SyntaxRegister::calculateComplexityScore)
+				.max()
+				.orElseThrow(); // a syntax info should have at least one pattern
+	}
+
+	private static int calculateComplexityScore(String pattern) {
+		int score = 0;
+		char[] chars = pattern.toCharArray();
+		for (int i = 0; i < chars.length; i++) {
+			if (chars[i] == '%') {
+				if (i - 2 >= 0 && chars[i - 2] == '%') { // weigh "%thing% %thing%" heavier
```

Comment:
perhaps strip all the [] and () chars first so `%thing% [%thing%]` is also caught. probably would need to string tag info first too.

Reply:
For efficiency/to prevent making it too much heavier, I slightly adjusted the check to consider two/three characters behind which should cover the case you mention